autoprecompiles/src/empirical_constraints.rs

```diff
@@ -88,6 +116,99 @@ fn merge_maps<K: Ord, V: Eq + Debug>(map1: &mut BTreeMap<K, V>, map2: BTreeMap<K
     }
 }
 
+type ConstraintList<A> = Vec<SymbolicConstraint<<A as Adapter>::PowdrField>>;
+
+/// For any program line that was not executed at least this many times in the traces,
+/// discard any empirical constraints associated with it.
+const EXECUTION_COUNT_THRESHOLD: u64 = 100;
+
+pub fn add_empirical_constraints<A: Adapter>(
+    empirical_constraints: &EmpiricalConstraints,
+    subs: &[Vec<u64>],
+    block: &BasicBlock<A::Instruction>,
+    columns: impl Iterator<Item = AlgebraicReference>,
+) -> (ConstraintList<A>, ConstraintList<A>) {
+    // Apply execution count threshold to avoid overfitting on rarely executed code
+    let empirical_constraints = empirical_constraints
+        .clone()
+        .with_thresholded_pc_count(EXECUTION_COUNT_THRESHOLD);
```

Comment:
This should avoid above-average effectiveness for the "other" category in the effectiveness plot. In fact, I would expect them to be below-average now, because they don't get any empirical constraints if they aren't executed at least 100 times.